File: 2024/java/softeer/통근버스_출발_순서_검증하기.java
import java.io.*;
import java.util.*;

public class Main {
    static int n;
    static int[] nums;
    static int bigger;
    static long ans;

    public static void main(String[] args) throws IOException{
        // input
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        n = Integer.parseInt(st.nextToken());

        nums = new int[n];
        st = new StringTokenizer(br.readLine());
        for(int i = 0 ; i < n; i++){
            nums[i] = Integer.parseInt(st.nextToken());
        }

        // 누적합
        for(int i = 0 ; i < n; i++) {
            bigger = 0;
            for (int k = i + 1; k < n; k++) {
                if (nums[i] < nums[k]) // 조건에 안맞음, a[i] 보다 큰 수라는 것만 체크
                    bigger += 1;
                else // 조건에 맞음 - 기존에 세었던 a[i] 보다 큰 값들 더해주기
                    ans += bigger;
            }
        }

        // output
        System.out.println(ans);
    }
}

File: 2024/python/softeer/통근버스_출발_순서_검증하기.py
import sys
n = int(input())
nums = list(map(int, input().split()))
ans = 0

for i in range(n):
    bigger = 0
    for k in range(i + 1, n):
        if nums[i] < nums[k]: # 조건에 안맞음, a[i] 보다 큰 수라는 것만 체크
            bigger += 1
        else: # 조건에 맞음 - 기존에 세었던 a[i] 보다 큰 값들 더해주기
            ans += bigger

print(ans)